components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -253,6 +253,22 @@ fn sqrt(arg: &Option<Real>) -> Result<Option<Real>> {
     }))
 }
 
+#[inline]
+#[rpn_fn]
+pub fn exp(arg: &Option<Real>) -> Result<Option<Real>> {
+    match arg {
+        Some(x) => {
+            let ret = x.exp();
+            if ret.is_infinite() || ret.is_nan() {
```

Comment:
Please handle overflow with an EvalContext

Reply:
It doesn't work, and break push down test, here is a similar example https://github.com/tikv/tikv/pull/5873#issuecomment-553829633